Add codespell support (config, workflow to detect/not fix) and make it fix some typos

yarikoptic opened the pull request:
More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

there might still be a few false positives to catch since overall a good number of fixes

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
Add github action to codespell hovudstraum on push and PRs

yarikoptic committed:
Ignore various 'interesting' words from codespell

ignore camelCase123

yarikoptic committed:
add inline ignore for lines with regexes or shell constructs

yarikoptic committed:
Fix typo manually

yarikoptic committed:
Ignore shortcut definitions like "[r]equire" and nin

yarikoptic committed:
[DATALAD RUNCMD] chore: run codespell throughout fixing a few typos interactively

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w -i 3 -C 4",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
more ignores of odd name choices

yarikoptic commented on .github/workflows/codespell.yml:
```diff
@@ -0,0 +1,25 @@
+  push:
+    branches: [hovudstraum]
+  pull_request:
+    branches: [hovudstraum]
```
or should apply to all?

Scotsguy commented on copyparty/web/browser.js:
```diff
@@ -4142,7 +4142,7 @@ var fileman = (function () {
 		var ire = ebi('rn_re'),
 			ifmt = ebi('rn_fmt'),
 			ipre = ebi('rn_pre'),
-			idel = ebi('rn_pdel'),
```
false positive

yarikoptic replied:
excluding

Scotsguy commented on copyparty/cert.py:
```diff
@@ -23,7 +23,7 @@ def ensure_cert(log: "RootLogger", args) -> None:
     """
     the default cert (and the entire TLS support) is only here to enable the
     crypto.subtle javascript API, which is necessary due to the webkit guys
-    being massive memers (https://www.chromium.org/blink/webcrypto)
```
false positive

yarikoptic replied:
that's cool -- I didn't even know about such modern word ;) excluding!

yarikoptic replied:
I force pushed with this excluded... not sure why here doesn't say "outdated".
<img width="1058" height="156" alt="image" src="https://github.com/user-attachments/assets/c5a07912-8d3e-4282-8758-acaa0531177e" />

Scotsguy commented on docs/versus.md:
```diff
@@ -259,7 +259,7 @@ symbol legend,
   * you can successfully play `$'\355\221'` with mpv through mounting a remote copyparty server with rclone, pog
 * `a`/copyparty remarks:
   * extremely minimal samba/cifs server
-  * netscape 4 / ie6 support is mostly listed as a joke altho some people have actually found it useful ([ie4 tho](https://user-images.githubusercontent.com/241032/118192791-fb31fe00-b446-11eb-9647-898ea8efc1f7.png))
```
altho -> although but not tho -> though?

yarikoptic replied:
;-) nice observation.  up to you -- we could exclude `altho` for symmetry or I could suggest `tho -> though` to codespell and fix here. Your desire?

Scotsguy replied:
it's not my decision to make in the end, either :) I'm just some guy, you know
(I like the casual tone of altho, if it matters)

Scotsguy commented on docs/changelog.md:
why change the changelog retroactively?

yarikoptic replied:
it is still a part of the "documentation" and I do not see why to keep it "buggy". But it is easy to skip if you insist.

Scotsguy replied:
It's also I think copied from https://github.com/9001/copyparty/releases for each release and therefore would get out of sync with those. it's probably not that big of a deal, tho

Scotsguy reviewed (changes requested):
A bunch of these are imo differences of style and not typos, some false positives too

yarikoptic committed:
[DATALAD RUNCMD] chore: run codespell throughout fixing a few  new typos automagically

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^
